interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -1724,8 +1726,8 @@ private SoakIterationResult performOneSoakIteration(
   }
 
   /**
-    * Runs large unary RPCs in a loop with configurable failure thresholds
-    * and channel creation behavior.
+   * Runs large unary RPCs in a loop with configurable failure thresholds
+   * and channel creation behavior.
    */
   public void performSoakTest(
```

Comment:
Rather than take a `resetChannelPerIteration` parameter, I think we can clean things up by taking a function parameter here instead.

The signature of the function param can be like this: `ManagedChannel maybeCreateNewChannel(ManagedChannel channel)`

In the rpc soak test, we can provide a callback that always returns the same channel.

In the channel soak test, we can provide a callback that closes the pass-in channel and returns a new one.